Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift

```diff
@@ -284,6 +292,90 @@ public final class ViewAnnotationManager {
         observers.removeValue(forKey: ObjectIdentifier(observer))
     }
 
+    // MARK: Framing
+
+    /// Calculates ``CameraOptions`` to fit the list of view annotations.
+    ///
+    /// - Important: This API isn't supported by Globe projection.
+    ///
+    /// - Parameter ids: The list of annotations ids to be framed.
+    /// - Parameter padding: See ``CameraOptions/padding``.
+    /// - Parameter bearing: See ``CameraOptions/bearing``.
+    /// - Parameter pitch: See ``CameraOptions/pitch``.
+    public func camera(for ids: [String], padding: UIEdgeInsets = .zero, bearing: CGFloat? = nil, pitch: CGFloat? = nil) -> CameraOptions? {
+        let options = ids.compactMap { try? mapboxMap.options(forViewAnnotationWithId: $0) }
+        guard !options.isEmpty else { return nil }
+
+        var north, east, south, west: CLLocationDegrees!
+        var accumulatedPadding = UIEdgeInsets.zero
+
+        for annotationOption in options where annotationOption.visible != false {
+            guard case .point(let point) = annotationOption.geometry else { continue }
+
+            let annotationFrame = annotationOption.frame
+            if north == nil || north > point.coordinates.latitude {
+                north = point.coordinates.latitude
+                accumulatedPadding.top =  padding.top + abs(annotationFrame.minY)
+            }
+            if east == nil || east < point.coordinates.longitude {
+                east = point.coordinates.longitude
+                accumulatedPadding.right = padding.right + annotationFrame.maxX
+            }
+            if south == nil || south < point.coordinates.latitude {
+                south = point.coordinates.latitude
+                accumulatedPadding.bottom = padding.bottom + annotationFrame.maxY
+            }
+            if west == nil || west > point.coordinates.longitude {
+                west = point.coordinates.longitude
+                accumulatedPadding.left = padding.left + abs(annotationFrame.minX)
+            }
+        }
+
+        let points = MultiPoint([
+            CLLocationCoordinate2D(latitude: north, longitude: east),
+            CLLocationCoordinate2D(latitude: south, longitude: west),
+        ])
+        return mapboxMap.camera(for: .multiPoint(points), padding: accumulatedPadding, bearing: bearing, pitch: pitch)
+    }
+
+    /// Sets the visible region so that the map displays the specified annotations.
+    ///
+    /// - Important: This API isn't supported by Globe projection.
```

Comment:
Also better to check if globe is used and throw as exception. i am planning to do the same in android

Reply:
[MapboxMap/camera(for:padding:bearing:pitch)](https://github.com/mapbox/mapbox-maps-ios/blob/main/Sources/MapboxMaps/Foundation/MapboxMap.swift#L412) does not throw an exception for Globe projection, it just returns a rather inaccurate camera, I'd rather put a warning log here instead of throwing exception because of consistency.